File: lab2/TOH.py
#tower of hanoi
# python treats global varibale as local unless expicitely declared as "global"

moves = 0

def TOH(n, src, dst, temp):
    global moves
    if n == 1:
        print (f"move disc{n} from {src} to {dst}\n")
        moves +=1
    else:
        global moves
        TOH(n-1, src, temp, dst)
        print (f"Move disc {n} from {src} to dst\n")
        TOH(n-1, temp, dst, src)
        moves += 1


n = int(input("Enter the number of plates"))
TOH(n, 'A', 'B', 'C')
print (f"The total number of moves required was {moves}")

File: lab2/TOH.cpp
//implementation of tower of hanoi
#include <iostream>
using namespace std;

int moves = 0;

void TOH(int n,char src,char dst,char temp)
{
    if (n==1)
    {
        cout << " Move disc " << n << " from " << src << " to " << dst << endl;
        moves++;
    }
    else
    {
        TOH(n - 1, src, temp, dst);
        cout << "Move disc " << n << " from " << src << " to " << dst << endl;
        TOH(n - 1, temp, dst, src);
        moves++;
    }
    
}

int main()
{
    
    int n;
    char src, dst, temp;
    TOH(3, 'A', 'C', 'B');
    cout << "The total number of moves = " << moves << endl;
    return 0;
}